### PR TITLE
refactor hashbydestination to a separate function

### PR DESCRIPTION
**Pull request by ramaraochavali**
- [ ] Configuration Infrastructure
- [ ] Docs
- [ ] Installation
- [X] Networking
- [ ] Performance and Scalability
- [ ] Policies and Telemetry
- [ ] Security
- [ ] Test and Release
- [ ] User Experience
- [ ] Developer Infrastructure

### CODE REVIEW

**Comment by ramaraochavali** [comment]
factored from https://github.com/istio/istio/pull/40401

File: pilot/pkg/networking/core/v1alpha3/route/route.go [hunk 1]
```diff
@@ -102,30 +104,12 @@ func BuildSidecarVirtualHostWrapper(routeCache *Cache, node *model.Proxy, push *
 
 	// dependentDestinationRules includes all the destinationrules referenced by the virtualservices, which have consistent hash policy.
 	dependentDestinationRules := []*model.ConsolidatedDestRule{}
-	// consistent hash policies for the http route destinations
-	hashByDestination := map[*networking.HTTPRouteDestination]*networking.LoadBalancerSettings_ConsistentHashLB{}
```
Comment (ramaraochavali) [inline pilot/pkg/networking/core/v1alpha3/route/route.go:106]: Main change is this -> refactor this in to separate function and reuse it.

File: pilot/pkg/networking/core/v1alpha3/route/route.go [hunk 2]
```diff
@@ -211,14 +195,14 @@ func separateVSHostsAndServices(virtualService config.Config,
 	return hosts, servicesInVirtualService
 }
 
-// buildSidecarVirtualHostsForVirtualService creates virtual hosts corresponding to a virtual service.
+// virtualHostsForVirtualService creates virtual hosts corresponding to a virtual service.
 // Called for each port to determine the list of vhosts on the given port.
 // It may return an empty list if no VirtualService rule has a matching service.
-func buildSidecarVirtualHostsForVirtualService(
+func virtualHostsForVirtualService(
```
Comment (hzxuzhonghu) [inline pilot/pkg/networking/core/v1alpha3/route/route.go:201]: verbs prefix

File: pilot/pkg/networking/core/v1alpha3/route/route.go [hunk 3]
```diff
@@ -266,7 +250,7 @@ func buildSidecarVirtualHostsForVirtualService(
 	return out
 }
 
-func buildSidecarVirtualHostsForService(
+func virtualHostsForService(
```
Comment (hzxuzhonghu) [inline pilot/pkg/networking/core/v1alpha3/route/route.go:253]: prefer previous name

File: pilot/pkg/networking/core/v1alpha3/route/route.go [hunk 4]
```diff
@@ -1259,7 +1243,7 @@ func consistentHashToHashPolicy(consistentHash *networking.LoadBalancerSettings_
 	return nil
 }
 
-func getHashForService(node *model.Proxy, push *model.PushContext,
+func hashForService(node *model.Proxy, push *model.PushContext,
```
Comment (hzxuzhonghu) [inline pilot/pkg/networking/core/v1alpha3/route/route.go:1246]: same

**Review by hzxuzhonghu — COMMENTED** [review]
minor comments

**Comment by ramaraochavali** [comment]
@hzxuzhonghu addressed all comments. PTAL when you get chance.

**Review by hzxuzhonghu — APPROVED** [review]
LGTM

**Pull request merged**